### ISSUE TITLE
Fullrender stops with IndexOutOfBoundsException

### ISSUE THREAD

**Issue by mr-ekkusu** [webbukkit/dynmap#2614]
Spigot 1.14.4
Dynmap-3.0-beta-5-spigot.jar

```
[22:14:04] [Dynmap Render Thread/INFO]: Full render of map 'flat' of 'world' in progress - 12600 tiles rendered (34.36 msec/tile, 64.03 msec per render)
[22:14:06] [Dynmap Render Thread/INFO]: Full render of map 'flat' of 'world' in progress - 12700 tiles rendered (34.23 msec/tile, 63.79 msec per render)
[22:14:09] [Dynmap Render Thread/INFO]: Full render of map 'flat' of 'world' in progress - 12800 tiles rendered (34.19 msec/tile, 63.69 msec per render)
[22:14:13] [Dynmap Render Thread/INFO]: Full render of map 'flat' of 'world' in progress - 12900 tiles rendered (34.26 msec/tile, 63.89 msec per render)
[22:14:17] [Dynmap Render Thread/INFO]: Full render of map 'flat' of 'world' in progress - 13000 tiles rendered (34.31 msec/tile, 64.00 msec per render)
[22:14:21] [Dynmap Render Thread/INFO]: Full render of map 'flat' of 'world' in progress - 13100 tiles rendered (34.34 msec/tile, 64.05 msec per render)
[22:14:25] [Dynmap Render Thread/INFO]: Full render of map 'flat' of 'world' in progress - 13200 tiles rendered (34.35 msec/tile, 64.11 msec per render)
[22:14:28] [Dynmap Render Thread/INFO]: Full render of map 'flat' of 'world' in progress - 13300 tiles rendered (34.36 msec/tile, 64.16 msec per render)
[22:14:31] [Dynmap Render Thread/ERROR]: [dynmap] Exception while fetching chunks:
java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
        at java.util.ArrayList.rangeCheck(ArrayList.java:657) ~[?:1.8.0_222]
        at java.util.ArrayList.set(ArrayList.java:448) ~[?:1.8.0_222]
        at net.minecraft.server.v1_14_R1.RegionFile.<init>(RegionFile.java:58) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.RegionFileCache.a(RegionFileCache.java:38) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.RegionFileCache.read(RegionFileCache.java:47) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at org.dynmap.bukkit.helper.v114_1.MapChunkCache114_1.loadChunkNBT(MapChunkCache114_1.java:316) ~[?:?]
        at org.dynmap.bukkit.helper.v114_1.MapChunkCache114_1.loadChunks(MapChunkCache114_1.java:407) ~[?:?]
        at org.dynmap.bukkit.DynmapPlugin$BukkitServer$6.call(DynmapPlugin.java:469) ~[?:?]
        at org.dynmap.bukkit.DynmapPlugin$BukkitServer$6.call(DynmapPlugin.java:456) ~[?:?]
        at org.bukkit.craftbukkit.v1_14_R1.scheduler.CraftFuture.run(CraftFuture.java:88) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at org.bukkit.craftbukkit.v1_14_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:394) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.MinecraftServer.b(MinecraftServer.java:1022) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.DedicatedServer.b(DedicatedServer.java:393) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.MinecraftServer.a(MinecraftServer.java:970) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:815) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_222]
[22:14:31] [Dynmap Render Thread/ERROR]: [dynmap] Exception while fetching chunks:
java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
        at java.util.ArrayList.rangeCheck(ArrayList.java:657) ~[?:1.8.0_222]
        at java.util.ArrayList.set(ArrayList.java:448) ~[?:1.8.0_222]
        at net.minecraft.server.v1_14_R1.RegionFile.<init>(RegionFile.java:58) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.RegionFileCache.a(RegionFileCache.java:38) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.RegionFileCache.read(RegionFileCache.java:47) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at org.dynmap.bukkit.helper.v114_1.MapChunkCache114_1.loadChunkNBT(MapChunkCache114_1.java:316) ~[?:?]
        at org.dynmap.bukkit.helper.v114_1.MapChunkCache114_1.loadChunks(MapChunkCache114_1.java:407) ~[?:?]
        at org.dynmap.bukkit.DynmapPlugin$BukkitServer$6.call(DynmapPlugin.java:469) ~[?:?]
        at org.dynmap.bukkit.DynmapPlugin$BukkitServer$6.call(DynmapPlugin.java:456) ~[?:?]
        at org.bukkit.craftbukkit.v1_14_R1.scheduler.CraftFuture.run(CraftFuture.java:88) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at org.bukkit.craftbukkit.v1_14_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:394) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.MinecraftServer.b(MinecraftServer.java:1022) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.DedicatedServer.b(DedicatedServer.java:393) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.MinecraftServer.a(MinecraftServer.java:970) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:815) ~[spigot.jar:git-Spigot-94af569-a3fdef8]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_222]
```

After that, dynmap never renders anything: nor full, nor update queue.
Even though they are mentioned as "ACTIVE" on /dynmap pause.

**Comment by baconwaifu** [comment]
It would be really helpful if dynmap actually spat out what chunk it had a problem fetching...

It appears that your world may be (partly) corrupt, given that it's minecraft code throwing an exception when reading some world data. If you can figure out what chunk it's complaining about, try visiting that chunk in-game, and see if it crashes or errors. If it does, you may be able to regenerate that portion of the world by just removing the relevant file.

**Comment by mr-ekkusu** [comment]
Could be related to #2636 and #2503

**Comment by FedUpWith-Tech** [comment]
I believe this issue is fixed and was specific to 1.14.4 and the beta versions of dynmap 3.0 if its still present in a current issue please re-open or create a new issue
